app.rb

```diff
@@ -0,0 +1,123 @@
+require './book'
+require './classroom'
+require './rental'
+require './person'
+require './student'
+require './teacher'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @class = Classroom.new('class 8')
+  end
+
+  def get_user_option(option)
+    case option
+    when '1'
+      all_books
+    when '2'
+      all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      rentals_by_id
+    else
+      puts 'Wrong option'
+      puts '--------------------------------------'
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def all_books
+    puts 'There are no books! Add a new book.' if @books.empty?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 1
+    puts
+  end
+
+  def all_people
+    puts 'The are no people! Please add a student or teacher.' if @people.empty?
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, Id: #{person.id}, Age: #{person.age}, " }
+    sleep 1
+    puts
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]:'
+    input = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    case input
+    when 1
+      create_student(age, name)
+    when 2
+      create_teacher(age, name)
+    end
+    puts 'Person created successfully!'
+  end
```

Comment:
- [ ] Following the `Single responsibility rule`, all requests for input and input validation should placed in classes/methods whose responsibilities are to request or validate input. Please checkout this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) and kindly fix that.